export const pTimeoutException = Symbol();

export function pTimeout<T>(
	promise: Promise<T>,
	time: number,
	exception: unknown = pTimeoutException
): Promise<T> {
	let timer: ReturnType<typeof setTimeout>;
	return Promise.race([
		promise,
		new Promise((_r, rej) => (timer = setTimeout(rej, time, exception)))
	]).finally(() => clearTimeout(timer)) as Promise<T>;
}

export * from './axios';
export * from './utils';
